utoipa-gen/src/path/media_type/encoding.rs

```diff
@@ -0,0 +1,92 @@
+use proc_macro2::{Ident, TokenStream};
+use quote::{quote, ToTokens};
+use syn::parse::{Parse, ParseStream};
+use syn::{parenthesized, Error, LitBool, LitStr, Token};
+
+use crate::parse_utils;
+
+// (content_type = "...", explode = true, allow_reserved = false,)
+#[derive(Default)]
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub struct Encoding {
+    pub(super) content_type: Option<String>,
```

Comment:
This could be `parse_utils::LitStrOrExpr`. This allows users to use a valid rust expression instead just literal str inlined to the macro.